src/main/kotlin/com/autonomousapps/model/declaration/Bucket.kt

```diff
@@ -4,12 +4,10 @@ package com.autonomousapps.model.declaration
 internal enum class Bucket(val value: String) {
   API("api"),
   IMPL("implementation"),
+  // These configurations go into the compileOnly bucket: '...CompileOny', '...CompileOnlyApi', 'providedCompile'
   COMPILE_ONLY("compileOnly"),
```

Comment:
This probably changes the meaning of what `Bucket` is supposed to mean here. I assume it is meant to be a 1-to-1 relationship between a _Bucket_ and a _Configuration_.
Adding three Buckets here, would mean three buckets that are treated the same in a bunch of other places. Could also be an option but it felt like a lot of copy/paste to me for this "small" improvement.

My thinking was to do it like this for now (compact solution covering what the plugin is capable right now) and adjust it more later if a distinction between "compileOnly" and "compileOnlyApi" becomes important.

For #648 (suggesting removals) all these three Configurations can be treated the same. So this implementation would work as a base for that.